Explorer/Assets/DCL/Multiplayer/SDK/Systems/WriteAvatarEmoteCommandSystem.cs

```diff
@@ -0,0 +1,91 @@
+using Arch.Core;
+using Arch.System;
+using Arch.SystemGroups;
+using Arch.SystemGroups.DefaultSystemGroups;
+using CRDT;
+using CrdtEcsBridge.ECSToCRDTWriter;
+using DCL.Diagnostics;
+using DCL.ECSComponents;
+using DCL.Multiplayer.SDK.Components;
+using DCL.Optimization.Pools;
+using ECS.Abstract;
+using ECS.LifeCycle.Components;
+using SceneRunner.Scene;
+
+namespace DCL.Multiplayer.SDK.Systems
+{
+    [UpdateInGroup(typeof(PresentationSystemGroup))]
+    [UpdateAfter(typeof(PlayerComponentsHandlerSystem))]
+    [LogCategory(ReportCategory.PLAYER_AVATAR_EMOTE_COMMAND)]
+    public partial class WriteAvatarEmoteCommandSystem : BaseUnityLoopSystem
+    {
+        private readonly ISceneStateProvider sceneStateProvider;
+        private readonly IECSToCRDTWriter ecsToCRDTWriter;
+        private readonly IComponentPool<PBAvatarEmoteCommand> componentPool;
+
+        public WriteAvatarEmoteCommandSystem(World world, IECSToCRDTWriter ecsToCRDTWriter, IComponentPool<PBAvatarEmoteCommand> componentPool, ISceneStateProvider sceneStateProvider) : base(world)
+        {
+            this.ecsToCRDTWriter = ecsToCRDTWriter;
+            this.componentPool = componentPool;
+            this.sceneStateProvider = sceneStateProvider;
+        }
+
+        protected override void Update(float t)
+        {
+            HandleComponentRemovalQuery(World);
+            UpdateAvatarEmoteCommandQuery(World);
+            CreateAvatarEmoteCommandQuery(World);
+        }
+
+        [Query]
+        [None(typeof(PBAvatarEmoteCommand), typeof(DeleteEntityIntention))]
+        private void CreateAvatarEmoteCommand(in Entity entity, ref PlayerSDKDataComponent playerSDKDataComponent)
+        {
+            if (playerSDKDataComponent.PlayingEmote == null) return;
```

Comment:
There is method `IsNullOrEmpty` for `URN`

Reply:
Replaced by `emoteCommand.PlayingEmote.IsNullOrEmpty()` checks